src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -315,7 +315,9 @@ internal virtual bool IsExplicitInterfaceImplementation
         /// Indicates whether the method is effectively readonly,
         /// by either the method or the containing type being marked readonly.
         /// </summary>
-        internal bool IsEffectivelyReadOnly => (IsDeclaredReadOnly || ContainingType?.IsReadOnly == true) && MethodKind != MethodKind.Constructor;
+        internal bool IsEffectivelyReadOnly => (IsDeclaredReadOnly || ContainingType?.IsReadOnly == true) && CouldBeReadOnly;
+
+        protected bool CouldBeReadOnly => !IsStatic && ContainingType.IsStructType() && MethodKind != MethodKind.Constructor;
```

Comment:
Consider: `IsValidReadOnlyTarget`. Think that may be more in line with other names.

Reply:
Done